anvil/src/input_handler.rs

```diff
@@ -172,7 +174,7 @@ impl<Backend> AnvilState<Backend> {
         // subsurface menus (for example firefox-wayland).
         // see here for a discussion about that issue:
         // https://gitlab.freedesktop.org/wayland/wayland/-/issues/294
-        if !pointer.is_grabbed() && !keyboard.is_grabbed() {
+        if !pointer.is_grabbed() && (!keyboard.is_grabbed() || input_method.keyboard_grabbed()) {
```

Comment:
Could you explain this `if`? Existing comment is already kinda big, but it looks like input_method keyboard grab is some sort of special case, and I find it hard to understand. Perhaps update the comment to include explanation of this.

EDIT: Or perhaps this is just a typo and `!` should be before bracket?

Reply:
Ah, no you are correct. Input methods are special cases, they are a replacement for the regular keyboards but they just redirect keyboard input to a client(the input method client). Focus is updated for text input, then redirected to the original keyboard surface handler. Input methods do not change surface handling in any way.

Maybe I misunderstood some detail about how this grab is supposed to work?

Reply:
Updated comment